swift/downgrades/04ef9ecaa9e06a587f252f147462861e3d32846a/downgrade.ql

```diff
@@ -0,0 +1,47 @@
+class Element extends @element {
+  string toString() { none() }
+}
+
+query predicate unspecified_elements(Element e, string property, string error) {
+  unspecified_elements(e, property, error)
+  or
+  error = "Removed parameter packs during the database downgrade" and
+  property = "" and
+  (
+    pack_element_exprs(e, _) or
+    pack_expansion_exprs(e, _) or
+    pack_element_types(e, _) or
+    pack_expansion_types(e, _, _) or
+    pack_types(e) or
+    element_archetype_types(e)
+  )
+}
+
+query predicate unspecified_element_children(Element e, int index, Element child) {
+  unspecified_element_children(e, index, child)
+  or
+  pack_element_exprs(e, child) and index = 0
+  or
+  pack_expansion_exprs(e, child) and index = 0
+  or
+  pack_element_types(e, child) and index = 0
+  or
+  (
+    pack_expansion_types(e, child, _) and index = 0
+    or
+    pack_expansion_types(e, _, child) and index = 1
+  )
+  or
+  pack_type_elements(e, index, child)
```

Comment:
children are needed for AST entities, so not for types. The `expr` ones look good though!

Reply:
Right, I somehow didn't think deeply about types' case.